attachments/views.py

```diff
@@ -220,17 +221,50 @@ def post(self, request, *args, **kwargs):
         except (InvalidExtensionException, InvalidFileTypeException, FileSizeException) as ex:
             return JsonResponse({'ok': False, 'error': force_text(ex)}, content_type=self.content_type)
         except VirusFoundException as ex:
-            return self.handle_virus_found(upload)
+            return self.handle_virus_found(upload, ex)
         except ValidationError as ex:
             return JsonResponse({'ok': False, 'error': force_text(ex.message)}, content_type=self.content_type)
-        except Exception as ex:
+        except Exception:
             logger.exception(f'Error attaching file to session {self.kwargs["session_id"]}')
             # Since this is a catch all we need to return a generic error (in case a more sensitive error occurred)
             return JsonResponse(
                 {'ok': False, 'error': 'An error occurred attaching file to the session.'},
                 content_type=self.content_type,
             )
 
+    @property
+    def file_is_zipped(self):
+        """Return True if self.file is a zip archive."""
+        return Path(self.file.name).suffix.lower() == ".zip"
+
+    def post(self, request, *args, **kwargs):
+        # Old versions of IE doing iframe uploads would present a Save dialog on JSON responses.
+        content_types = {'IFrame': 'text/plain'}
+        self.content_type = content_types.get(request.POST.get('X-Requested-With'), 'application/json')
+        zip_error_msg = ""
+        files = [self.file]
+        errors = []
+        if self.session.unpack_zip_files and self.file_is_zipped:
+            zip_error_msg = f"Error(s) in {self.file.name}:<br /><br />"
+            with ZipFile(self.file, "r") as zip_file:
+                files = [
+                    SimpleUploadedFile(name=name, content=BytesIO(zip_file.read(name)).getbuffer())
+                    for name in zip_file.namelist()
+                ]
+            if not files:
+                errors.append("Zip file is empty.")
+        for self.file in files:
+            if self.session.unpack_zip_files and self.file_is_zipped:
+                errors.append(f"{self.file.name} - Error: nested zip files are not supported.")
+            else:
+                json_response = self.handle_file_upload(request, *args, **kwargs)
+                if error := json.loads(json_response.content).get("error", None):
+                    errors.append(error)
+        if errors:
+            err_msg = zip_error_msg + "<br />".join(errors)
+            return JsonResponse({"ok": False, "error": err_msg}, content_type=self.content_type)
+        return json_response
```

Comment:
This json_response will be the json_response of the last file in the zip file, right?  I'm trying to think if there are any negative consequences of that or if we should build a seperate json response if the zip was unpacked?

Reply:
Yes, that's correct. I didn't run into any problems when I was testing it, but it does feel a little weird.

The json_response that would be returned in this case looks like this:
```
            JsonResponse(
                {'ok': True, 'file_name': upload.file_name, 'file_size': upload.file_size},
                content_type=self.content_type,
            )
```

But I don't think we're ever doing anything with `file_name` and `file_size`, are we? Could we just drop those keys when this JsonResponse object is built in `handle_file_upload()`? And I don't mean just if it's a zip file...I mean always.

Reply:
Hmm my only concern is that *we* aren't using it, but as a 3rd party app we can't be know for sure if another site using this app is utilizing those key/value pairs.

I think in the event of unzipping, maybe we can add to the JSON response? So they key value pairs would be:

- ok - True/False
- file_name - the name of the zip file since that was the originally uploaded file
- file_size - the size of the zip
- files - (new) - an array of the dicts returned by `handle_file_upload` by each of the files

Thoughts on this? I think it maintains expected behavior, but gives you access to all information if you decide to opt-into the unzipping feature

Reply:
I like it! As always, I forgot that other people may be using this app...

I'll get these changes in soon.